julia-mode.el

```diff
@@ -705,10 +705,11 @@ Return nil if point is not in a function, otherwise point."
   ;; don't use syntax classes, screws egrep
   '(("Function (_)" "[ \t]*function[ \t]+\\(_[^ \t\n]*\\)" 1)
     ("Function" "^[ \t]*function[ \t]+\\([^_][^\t\n]*\\)" 1)
-    ("Const" "[ \t]*const \\([^ \t\n]*\\)" 1)
-    ("Type"  "^[ \t]*[a-zA-Z0-9_]*type[a-zA-Z0-9_]* \\([^ \t\n]*\\)" 1)
-    ("Require"      " *\\(\\brequire\\)(\\([^ \t\n)]*\\)" 2)
-    ("Include"      " *\\(\\binclude\\)(\\([^ \t\n)]*\\)" 2)
+    ("Function" "^[ \t]*\\(\\(@.+ \\)?[_a-zA-Z0-9]+\(.*\)\\) *= " 1)
+    ("Const" "^[ \t]*const \\([^ \t\n]*\\) *= " 1)
+    ("Struct" "^[ \t]*\\(\\(mutable \\)*struct [^ \t\n]*\\)" 1)
+    ("Require" " *\\(\\brequire\\)(\\([^ \t\n)]*\\)" 2)
```

Comment:
It's hard to tell exactly which lines/regexes were changed in this commit and which are just whitespace cleanups. Could you please separate out the pure whitespace cleanups into a separate commit? Or are they all changing, and my eyes are just failing me?

Reply:
Haha, I can see (or not?) where you're coming from.

All lines except `Require` & `Include` changed. `Require` and `Include` only contain a whitespace change.

I can try to add a short comment for each line on how the regex works, because I'm not sure looking at the individual commits is much help?

Reply:
It would help greatly if you could rewrite the changed regexps using `rx`. See https://github.com/JuliaEditorSupport/julia-emacs#contributing

Reply:
Rewrote to `rx` — I think this also makes the diff much clearer.

Reply:
Thanks for the rewrite, indeed they are much cleaner. I wonder if it would make sense to add some tests (basically just testing that the regexps match what they need to). I realize that the package currently does not have tests for these, but it would make it easier to maintain.

Reply:
I agree, however probably won't have time anytime soon to add them 🙁